api/http_internal.go

```diff
@@ -53,7 +53,7 @@ func ListenAndServeInternal(ctx context.Context, cli config.Cli, vodEngine *pipe
 	return server.Shutdown(ctx)
 }
 
-func NewCatalystAPIRouterInternal(cli config.Cli, vodEngine *pipeline.Coordinator, mapic mistapiconnector.IMac, bal balancer.Balancer, c cluster.Cluster) *httprouter.Router {
+func NewCatalystAPIRouterInternal(ctx context.Context, cli config.Cli, vodEngine *pipeline.Coordinator, mapic mistapiconnector.IMac, bal balancer.Balancer, c cluster.Cluster, broker misttriggers.Broker) *httprouter.Router {
```

Comment:
This is the only bit of the PR I don't love - the approach I'd probably take is just to YOLO the server shutdown case, since it's not like any of the handlers are going to be long-running processes. 

Another approach would be to maybe store the global context in some top-level variable and then derive a local context off that at the point we're about to loop through instantiating all the handler functions.